tidy3d/components/geometry.py

```diff
@@ -3335,6 +3500,207 @@ def _surface_area(self, bounds: Bound) -> float:
         return area
 
 
+class ComplexPolySlabBase(PolySlab):
+    """Interface for dividing a complex polyslab where self-intersecting polygon can
+    occur during extrusion. This class should not be used directly. Use instead
+    :class:`plugins.polyslab.ComplexPolySlab`."""
```

Comment:
`ComplexPolySlabBase` doesn't include the `to_structure` method because it would create cyclic dependencies. That's only defined for the plugin version, which is allowed to import any core component. So, to minimize confusion, I try to instruct the user to always use the plugin version, otherwise they might try to call `to_structure` and get an error.